RA-08-RecyclerView-realization

DenisGubernatorov opened the pull request:
Реализция адаптера для recyclerView + небольшой рефакторинг

DenisGubernatorov committed:
RA-07-categories-screen
1) Доработка кнопки "Избранное". Если текст не помещается в контейнер, то он обрезается и ставится многоточие
2) В ресурсы добавлены изображения для категорий

DenisGubernatorov committed:
Revert "RA-07-categories-screen"

This reverts commit 40a49518934f331974ef1029635b735c2dab2766.

DenisGubernatorov committed:
RA-08-RecyclerView-realization
1) Доработка кнопки "Избранное". Если текст не помещается в контейнер, то он обрезается и ставится многоточие
2) В ресурсы добавлены изображения для категорий

DenisGubernatorov committed:
Revert "RA-08-RecyclerView-realization"

This reverts commit 00b45dc4f11af9b11e6a57c970cd9c6f07dd2414.

DenisGubernatorov committed:
RA-08-RecyclerView-realization
1) Доработка кнопки "Избранное". Если текст не помещается в контейнер, то он обрезается и ставится многоточие
2) Переименование файлов с изображением для категорий
3) Реализован адаптер для recyclerView
4) Фон карточки сделан как на макете, белым. Так как на телефоне с темной темой, он тоже был черным

DenisGubernatorov commented:
Опять сюрприз, когда открыл приложение на реальном телефоне. Картинки в карточках категорий меньше чем на макете и в эмуляторе 
![image](https://github.com/user-attachments/assets/f84c8c89-5b62-4895-b292-65f8e8b07487)
Поигрался с высотой, менял наследование, веса выставить. Что-то такой вариант, как на скрине - самый лучший из того что получалось

aGroupoid commented on app/src/main/java/com/example/recipesapp/CategoriesListAdapter.kt:
```diff
@@ -0,0 +1,45 @@
+                null
+            }
+            binding.cardImageView.setImageDrawable(drawableImage)
+
```
Надо еще добавить “описание контента”. contentDescription используется для предоставления текстового описания для не текстовых элементов, таких как изображения, кнопки с иконками и других графических объектов.
Это происходит так: пользователь водит пальцем по экрану и в зависимости от того, где сейчас активно нажатие – происходит озвучивание. Чтобы перейти в конкретную категорию пользователю нужно совершить двойное нажатие в качестве подтверждения перехода на соответствующую категорию.

Для нединамических фото, достаточно прописать в XML imageView.
Для карточек, менять его в адаптере. Можно задавать картинкам категорий что-то "Изображение категории название_категории".

DenisGubernatorov replied:
В сообщении коммита написал, что для того, что бы хорошо было видно riple убрал настройку бэкграунда в лэйауте. Но тогда карточка ужасно смотрится если установлена темная тема. Но это наверное больше вопрос к макету и его реализации в светлой и темной темах
![image](https://github.com/user-attachments/assets/9c1ea608-ed22-430b-b817-b665325b92a2)

aGroupoid replied:
Не совсем понимаю про какой бэкграунд идёт речь. На темной теме, 30 СДК, у меня она хорошо работает. Специфическое поведение твоего девайса и версии видимо
![image](https://github.com/user-attachments/assets/293c7eb8-de7a-449b-b023-32eb3ef3a209)

aGroupoid commented on app/src/main/java/com/example/recipesapp/CategoriesListFragment.kt:
```diff
@@ -20,11 +21,20 @@ class CategoriesListFragment : Fragment() {
+        val categoriesListAdapter = CategoriesListAdapter(STUB.getCategories())
+        val rvCategories = binding.rvCategories
+        rvCategories.adapter = categoriesListAdapter
+        binding.rvCategories.layoutManager = GridLayoutManager(requireContext(), 2)
```
Эти параметры уже прописаны в xml. Дублировать здесь их не надо

aGroupoid commented on app/src/main/res/layout/item_category.xml:
```diff
@@ -12,29 +12,33 @@
             android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-
+            android:layout_height="match_parent"
```
Давай лучше просто захардкодим длину картинки, как на дизайне. На моём устройсте с таким подходом фотка вообще мелкая

DenisGubernatorov replied:
Кстати, при повороте телефона для карточек остается "амбразура". Скрин из менеджера запущенных устройств, но на телефоне такие пропорции... Видимо это опять вопрос к макету и к тому, что должен быть вариант компановки для таких случаев :))
![image](https://github.com/user-attachments/assets/1d8fcde0-23d1-4d15-99f2-99b89b949770)

aGroupoid replied:
Да, это тестовый макет. На реальном приложении был бы макет для другой ориентации, либо пришлось бы пинать дизайнера, чтоб сделал.  Как вариант, могу предложить шапку внести в scrollView, так проблем особо не будет

aGroupoid reviewed (changes requested):
При нажатии на элементы списка - нет никакого эффекта. Добавь cardView эти параметры, чтоб появился ripple
https://gist.github.com/mrsid96/1ec250e793ad914de27722bdda52e0b1

DenisGubernatorov committed:
RA-08-RecyclerView-realization
1) Повышение версий библиотек, согласно предупреждениям от идешки
2) Добавлено описание изображений в категориях для talkback
3) Убрана лишняя программная инициализация лэйаут менеджера
4) Размер картинок в категориях сделан фиксированным
5) Доработка CardView для анимации нажатия. Убрана установка background в linerarlayout, так как атрибут передавливал настройки анимации нажатия в cardView

aGroupoid reviewed (approved):
Молодец, всё верно!
